core/clusters/clusters.go

```diff
@@ -875,9 +886,51 @@ func scaleDep(dcc oclient.DeploymentConfigInterface, clustername string, count i
 
 	// if err has a value here then we failed all retries and err is the last message from a failed update
 	if err != nil {
-		return generalErr(err, fmt.Sprintf(updateDepMsg, otype), ClientOperationCode)
+		return updated, generalErr(err, fmt.Sprintf(updateDepMsg, otype), ClientOperationCode)
 	}
-	return nil
+	return updated, nil
+}
+
+func updateAnnotation(dcc oclient.DeploymentConfigInterface, clustername string, wup bool, mup bool, workers int, masters int) error {
+	var updateerr error
+	for i := 0; i < 20; i++ {
+		master, err := dcc.Get(mastername(clustername))
+		if err == nil {
+			// Annotations have to be filled in because we did it on create
+			// but check anyway
+			ann := master.GetAnnotations()
+			if ann != nil {
+				cc := ClusterConfig{}
+				err := json.Unmarshal([]byte(ann["oshinko-config"]), &cc)
+				if err == nil {
+					if wup {
+						cc.WorkerCount = workers
+					}
+					if mup {
+						cc.MasterCount = masters
+					}
+					configbytes, err := json.Marshal(cc)
+					if err == nil {
+						ann["oshinko-config"] = string(configbytes)
+						master.SetAnnotations(ann)
+						_, updateerr = dcc.Update(master)
+						if updateerr == nil {
+							// If it worked, get out of here
+							break
+						}
+					}
+				}
+			} else {
+				// No annotation, so nothing to updated
+				break
+			}
+		}
+		if err != nil {
+			return err
+		}
+		time.Sleep(250*time.Millisecond)
```

Comment:
What is the sleep for?

Reply:
sleep is just so we don't spin wildly, it may not be necessary